### ISSUE TITLE
Parsing nil values in XML causing ArgumentError

### ISSUE THREAD

**Issue by sarsena** [mvz/happymapper#68]
When trying to parse an xml object that has an empty string`""` and doesn't meet the `nil?` criteria for a DateTime class I get `ArgumentError: invalid date`.  I noticed the original jnunemaker/happymapper had a [rescue](https://github.com/jnunemaker/happymapper/blob/master/lib/happymapper/item.rb#L109) in the method that perform the `parse`.  Is there any way we could possibly re-add that to the [supported_types class](https://github.com/dam5s/happymapper/blob/567156915da5750326e26c0df8afea635b1c5c5e/lib/happymapper/supported_types.rb#L100-L138)?  I can submit a PR to merge if you could review.  Or if there is already a way this is dealt with please let me know

```2) Error:
test_0001_returns the proper length(Summary.for_email::with cassette Summary.for_email):
ArgumentError: invalid date
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/supported_types.rb:113:in `parse'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/supported_types.rb:113:in `block in <module:SupportedTypes>'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/supported_types.rb:74:in `apply'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/item.rb:72:in `typecast'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/item.rb:95:in `process_node_as_supported_type'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/item.rb:38:in `block in from_xml_node'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/element.rb:20:in `find'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper/item.rb:38:in `from_xml_node'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:410:in `block (3 levels) in parse'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:409:in `each'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:409:in `block (2 levels) in parse'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:396:in `map'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:396:in `block in parse'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-1.6.8.1/lib/nokogiri/xml/node_set.rb:187:in `block in each'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-1.6.8.1/lib/nokogiri/xml/node_set.rb:186:in `upto'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-1.6.8.1/lib/nokogiri/xml/node_set.rb:186:in `each'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:394:in `each_slice'
    /opt/rubies/2.3.1/lib/ruby/gems/2.3.0/gems/nokogiri-happymapper-0.5.9/lib/happymapper.rb:394:in `parse'
    /Users/sarsena/src/order_client/lib/order_client/summary.rb:127:in `for_email'
    /Users/sarsena/src/order_client/test/cases/integration_test.rb:235:in `block (3 levels) in <class:IntegrationTest>'

XML:
    <summary>
         <order-id type="integer">1726827</order-id>
           <placed-at type="datetime">2011-09-20T12:42:13-04:00</placed-at>
            <cancelled-at type="datetime">2011-09-20T15:19:23-04:00</cancelled-at>
            <cancellation-reason type="string">INTERNAL</cancellation-reason>
            <delivered-at nil="true"></delivered-at>
            <shipped-at nil="true"></shipped-at>
```

**Comment by sarsena** [comment]
I have a forked version [here](https://github.com/sarsena/happymapper).  See [commit](https://github.com/sarsena/happymapper/commit/3c19d2528ed0b5ab47e077ad179f9b41dca3dadd)

**Comment by mvz** [comment]
This was fixed in #69, as far as I can see.
